pkg/auth/signedMessage.go

```diff
@@ -33,20 +33,72 @@ var (
 	Signer = errs.Class("Message public key did not match expected signer")
 	//BadID indicates a public key / node id mismatch
 	BadID = errs.Class("Node ID did not match expected id")
+
+	//Marshal indicates a failure during serialization
+	Marshal = errs.Class("Could not marshal item to bytes")
+	//Unmarshal indicates a failure during deserialization
+	Unmarshal = errs.Class("Could not unmarshal bytes to item")
+	//Missing indicates missing or empty information
+	Missing = errs.Class("Required field is empty")
 )
 
+//SignableMsg is a protocol buffer with a certs and a signature
+//Note that we assume proto.Message is a pointer receiver
+type SignableMsg interface {
+	proto.Message
+	GetCerts() [][]byte
+	GetSignature() []byte
+	SetCerts([][]byte) bool
+	SetSignature([]byte) bool
+}
+
+//SignMsg adds the crypto-related aspects of signed message
+func SignMsg(msg SignableMsg, ID identity.FullIdentity) error {
+	if msg == nil {
+		return Missing.New("message")
+	}
+	_ = msg.SetSignature(nil)
+	_ = msg.SetCerts(nil)
+	msgBytes, err := proto.Marshal(msg)
+	if err != nil {
+		return Marshal.Wrap(err)
+	}
+	privECDSA, ok := ID.Key.(*ecdsa.PrivateKey)
+	if !ok {
+		return ECDSA
+	}
+	signature, err := cryptopasta.Sign(msgBytes, privECDSA)
+	if err != nil {
+		return Sign.Wrap(err)
+	}
+	_ = msg.SetSignature(signature)
+	_ = msg.SetCerts(ID.ChainRaw())
+	return nil
+}
+
 //VerifyMsg checks the crypto-related aspects of signed message
-func VerifyMsg(sm pb.SignedMsg, signer storj.NodeID) error {
-	//no null fields
-	if ok, err := pb.MsgComplete(sm); !ok {
-		return err
+func VerifyMsg(msg SignableMsg, signer storj.NodeID) error {
+	//setup
+	if msg == nil {
+		return Missing.New("message")
+	} else if msg.GetSignature() == nil {
+		return Missing.New("message signature")
+	} else if msg.GetCerts() == nil {
+		return Missing.New("message certificates")
 	}
-	certs := sm.GetCerts()
+	signature := msg.GetSignature()
+	certs := msg.GetCerts()
+	_ = msg.SetSignature(nil)
+	_ = msg.SetCerts(nil)
+	msgBytes, err := proto.Marshal(msg)
```

Comment:
Don't assume that different versions of protobuf serialize into the same format. This will cause issues when upgrading to let's say protobuf 4.

Reply:
I'll save this for another PR.